src/main/kotlin/com/devfloor/untitled/article/application/ArticleRequest.kt

```diff
@@ -10,6 +10,8 @@ data class ArticleRequest(
     val content: String,
     val hashtags: List<String>
 ) {
+    val isOptionsNotEmpty: Boolean = options.isNotEmpty()
```

Comment:
이 로직은 사실 options가 변경될 여지가 거의 없어서 괜찮을 것 같은데 그래도 안전하게 작성하기 위해서 아래처럼 수정하는 건 어때?

```suggestion
    val isOptionsNotEmpty: Boolean
        get() = options.isNotEmpty()
```

이렇게 하면 매번 호출하는 시점에서 options의 상태에 대한 결과값을 받을 수 있어서 좀 더 안전할 것 같아. 사실 List는 객체라서 val options라도 리스트에 값을 추가하는 등의 작업은 가능한 걸로 알고 있어서 함수나 상태를 프로퍼티로 사용할 경우 이처럼 get을 통해서 처리하는 걸로 알고 있어.

Reply:
현재 options는 immutable 리스트라서 어차피 누군가 실수로 setter로 뭔가 수정작업을 시도하려 해도 수정지 되지 않기 때문에 그런 위험성은 막지않아도 이미 괜찮은거 아니야? 굳이 getter를 사용해야 하는 이유가 또 뭐가 있는거야??

Reply:
아 코틀린에서는 기본이 immutable이라 상관이 없긴 하겠네. 그런데 저렇게 로직을 가지는 프로퍼티에 getter로 사용하는 이유는

```kotlin
val isOptionsNotEmpty: Boolean = options.isNotEmpty()
```

위의 로직의 경우 ArticleRequest가 생성됨과 동시에 초기화가 돼. 그래서 사실 여기서는 문제가 발생하진 않지만 해당 프로퍼티가 가지는 의미는 **생성 시점에서의 options의 상태**라고 생각할 수 있을 것 같고

```kotlin
val isOptionsNotEmpty: Boolean
    get() = options.isNotEmpty()
```

위의 로직의 경우 해당 프로퍼티를 접근할 때마다 **조회 시점에서의 options의 상태**에 대한 조회가 이뤄져.

그래서 프로퍼티를 설정할 때 생성 시점에 대한 정보를 저장할 프로퍼티는 위 방법으로, 그리고 객체의 상태에 대한 조회가 목적인 프로퍼티의 경우 아래의 방법으로 사용한다고 알고 있어.

현재 로직상에서 options는 immutable이라서 문제가 발생할 여지는 없는 것 같아. 그런데 우리가 사용하는 목적이 생성 시점의 상태를 궁금해하기보다는 조회할 때마다의 상태를 궁금해하는 쪽에 더 가깝다고 생각이 들어서 아래 방법을 제안한거였어 ㅎㅅㅎ

Reply:
https://kotlinlang.org/docs/properties.html#getters-and-setters
아 그렇네 ! 함수가 하는 것 같은 일을 프로퍼티로 만들었을 때 custom getter로 접근하는게 맞는거같다!

Reply:
추가적으로 getter 반환 타입이 유추 가능하다고 생각하면 반환형도 생략해도 된다고 하네!